engine/table/src/main/java/io/deephaven/engine/table/impl/updateby/UpdateByOperatorFactory.java

```diff
@@ -1371,52 +1383,115 @@ private UpdateByOperator makeRollingFormulaOperator(@NotNull final MatchPair pai
             } else if (csType == long.class || csType == Long.class) {
                 return new LongRollingFormulaOperator(pair, affectingColumns,
                         rs.revWindowScale().timestampCol(),
-                        prevWindowScaleUnits, fwdWindowScaleUnits, rs.formula(), rs.paramToken(),
+                        prevWindowScaleUnits, fwdWindowScaleUnits, formula, paramToken,
                         formulaColumnMap, tableDef, compilationProcessor);
             } else if (csType == float.class || csType == Float.class) {
                 return new FloatRollingFormulaOperator(pair, affectingColumns,
                         rs.revWindowScale().timestampCol(),
-                        prevWindowScaleUnits, fwdWindowScaleUnits, rs.formula(), rs.paramToken(),
+                        prevWindowScaleUnits, fwdWindowScaleUnits, formula, paramToken,
                         formulaColumnMap, tableDef, compilationProcessor);
             } else if (csType == double.class || csType == Double.class) {
                 return new DoubleRollingFormulaOperator(pair, affectingColumns,
                         rs.revWindowScale().timestampCol(),
-                        prevWindowScaleUnits, fwdWindowScaleUnits, rs.formula(), rs.paramToken(),
+                        prevWindowScaleUnits, fwdWindowScaleUnits, formula, paramToken,
                         formulaColumnMap, tableDef, compilationProcessor);
             }
             return new ObjectRollingFormulaOperator<>(pair, affectingColumns,
                     rs.revWindowScale().timestampCol(),
-                    prevWindowScaleUnits, fwdWindowScaleUnits, rs.formula(), rs.paramToken(),
+                    prevWindowScaleUnits, fwdWindowScaleUnits, formula, paramToken,
                     formulaColumnMap, tableDef, compilationProcessor);
         }
 
+        private UpdateByOperator makeRollingFormulaMultiColumnOperator(
+                @NotNull final TableDefinition tableDef,
+                @NotNull final RollingFormulaSpec rs) {
+
+            final long prevWindowScaleUnits = rs.revWindowScale().getTimeScaleUnits();
+            final long fwdWindowScaleUnits = rs.fwdWindowScale().getTimeScaleUnits();
+
+            final Map<String, ColumnDefinition<?>> columnDefinitionMap = tableDef.getColumnNameMap();
+
+            // Create the colum
+            final SelectColumn selectColumn = SelectColumn.of(Selectable.parse(rs.formula()));
+
+            // Get or create a column definition map where the definitions are vectors of the original column types.
+            if (vectorColumnNameMap == null) {
+                vectorColumnNameMap = new HashMap<>();
+                columnDefinitionMap.forEach((key, value) -> {
+                    final ColumnDefinition<?> columnDef = ColumnDefinition.fromGenericType(
+                            key, VectorFactory.forElementType(value.getDataType()).vectorType());
```

Comment:
Add component type (that is, `value.getDataType()`). I wonder if that means we need more testing for Object inputs.

Reply:
This is done, thinking about add'l testing